#OM Namo Lakshmi Pataye Namaha
"""SourcePatternData

Automatically generated by Colab.

Original file is located at
    https://colab.research.google.com/drive/1dznT8iwBTjRajaIbZZZJaQNJFErrud7k
"""

error_raw_data={'''session log doesn't has any data''':[''],
    'DTM ERROR FOR COLUMNS':['''[DIRECTOR> TM_6279 The session instance [s_mtt_012BH30Z00000000KERF] encountered the following run-time validation error: [Lookup Transformation  has INVALID lookup condition: datatypes for ports "CHECK_2" and "COL_VAL" do not match.''',
                                     '''"DIRECTOR> TM_6279 The session instance [s_mtt_01DJCHBS78EQ97EYQEGRF] encountered the following run-time validation error: [Joiner Transformation  has INVALID join condition: datatypes for ports "CUSTOMER" and "EXECUTION" do not match.].''',
                                     '''DIRECTOR> TM_6279 The session instance [s_mtt_01JHBAZCKWHXOIJDKJJHABF] encountered the following run-time validation error: [Joiner Transformation  has INVALID join condition: datatypes for ports "VALUE" and "NUMBER" do not match.'''
            ],
            'Connection issue or Using wrong Secure Agent' : ['''Error connecting to database [
[Informatica][ODBC Oracle Wire Protocol driver]Connection refused. Verify Host Name and Port Number.[Informatica][ODBC Oracle Wire Protocol driver][Oracle]ORA-12203: unable to connect to destination
Failed to connect to database using user [abstpt] and connection string [Driver=DataDirect CLOSED 8.0 Oracle Wire Protocol for Informatica - BASE;Host=ajdcbcnsic.corp.pep.tst;ServiceName=djbxwsxn;Port=1550;UID=ywadgsbau;AS=6000000;EBL=0;ENS=1;ETWT=1;WorkArounds2=2;EnableEmptyLobs=1]
''',
'''Error connecting to database [
[Informatica][ODBC Oracle Wire Protocol driver]Connection refused. Verify Host Name and Port Number.[Informatica][ODBC Oracle Wire Protocol driver][Oracle]ORA-12203: unable to connect to destination
Failed to connect to database using user [abstpt] and connection string [Driver=DataDirect CLOSED 8.0 Oracle Wire Protocol for Informatica - BASE;Host=hostcloud;ServiceName=service2;Port=2580;UID=connverssion1;AS=6000000;EBL=0;ENS=1;ETWT=1;WorkArounds2=2;EnableEmptyLobs=1]''',
'''READER_2_1_1> RR_4036 [2025-08-29 15:28:14.893] Error connecting to database [ Database driver error... Database driver error... Database Error: Failed to connect to database using user [user_ver] and connection string [Driver=DataDirect CLOSED 8.0 Oracle Wire Protocol for Informatica - BASE;Host=cloudpet.corp.pep.tst;ServiceName=databasecheck;Port=1540;UID=user_ver;AS=6000000;EBL=0;ENS=1;ETWT=1;WorkArounds2=2;EnableEmptyLobs=1].]. READER_1_1_1> BLKR_16001 [2025-08-29 15:28:14.893] Error connecting to database... MAPPING> TM_6018 [2025-08-29 15:28:14.903] The session completed with [0] row transformation errors.''']
,

    'PM_PARSE: Each column in the lookup should me mentioned with as while using query' :['''MAPPING> TE_7017 Internal error. Failed to initialize transformation [lkp_flower]. Contact Informatica Global Customer Support. MAPPING> DBG_21173 ERROR : Lookup port CD_ID in lkp_flower not specified in lookup override''',
                                                                                          '''MAPPING> TE_7017 Internal error. Failed to initialize transformation [checking_col2]. Contact Informatica Global Customer Support. MAPPING> DBG_21173 ERROR : Lookup port CUSTUMER_CD in checking_col2 not specified in lookup override''',
                                                                                          '''MAPPING> TE_7017 Internal error. Failed to initialize transformation [DATA_TABLE]. Contact Informatica Global Customer Support. MAPPING> DBG_21173 ERROR : Lookup port VARI_MODE in DATA_TABLE not specified in lookup override''']
                ,
                'check lookup cache path':['''LKPDP_12> DBG_21045 ERROR: Can't create files in the directory "/root/path/folder2/CHECK/VALUE/Cache/"
LKPDP_12> TE_7017 Internal error. Failed to initialize transformation [variable_VAL_DATA{{BLD}}]. Contact Informatica Global Customer Support.''',
                                           '''LKPDP_2> DBG_21048 ERROR: Can't create files in the directory "/c:/batch/metal/color/"
LKPDP_12> TE_7017 Internal error. Failed to initialize transformation [LOC_YIPPIE{{BLD}}]. Contact Informatica Global Customer Support.''']}

patterns=['Since DTM Error Logging is enabled, row level errors will not be written to the session log.','The DTM Error Log File Delimiter being used ',
          '''defaultvalue=[ERROR('transformation error')]''','Row will be skipped if transformation errors are encountered',
          'will be used if transformation errors are encountered','Error initializing DTM for session',
          'Error occurred while running the Session','Cumulative Error Code','DTM error log','Using configuration property',
          'Property Name','<Warning>','The session completed with','Read name and value pair','error rows for source table','Finished transformations for']
class SourcePatternData():
  def source_data():
    return error_raw_data

  def patterns_data():
    return patterns

